cloud-sql/postgres/Startup.cs

```diff
@@ -45,47 +44,105 @@ public Startup(IConfiguration configuration)
         // http://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
-            string projectId = Google.Api.Gax.Platform.Instance().ProjectId;
-            if (!string.IsNullOrEmpty(projectId))
-            {
-                services.AddGoogleExceptionLogging(options =>
-                {
-                    options.ProjectId = projectId;
-                    options.ServiceName = "CloudSqlSample";
-                    options.Version = "Test";
-                });
-            }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddSingleton(sp => new NpgsqlConnection().GetPostgreSqlConnectionString());
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
-            _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env,
+            ILoggerFactory loggerFactory)
         {
-            DbConnection connection;
-            connection = GetPostgreSqlConnection();
-            connection.Open();
-            using (var createTableCommand = connection.CreateCommand())
+            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+
+            if (env.IsDevelopment())
             {
-                createTableCommand.CommandText = @"
-                    CREATE TABLE IF NOT EXISTS
-                    votes(
-                        vote_id SERIAL NOT NULL,
-                        time_cast timestamp NOT NULL,
-                        candidate VARCHAR(6) NOT NULL,
-                        PRIMARY KEY (vote_id)
-                    )";
-                createTableCommand.ExecuteNonQuery();
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                // Configure error reporting service.
+                app.UseGoogleExceptionLogging();
+                app.UseExceptionHandler("/Home/Error");
             }
-            return connection;
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+    }
+
+        static class StartupExtensions
+    {
+        public static DbConnection OpenWithRetry(this DbConnection connection) =>

```

Comment:
Same proposed changes to Polly config may be applid here, and then the ones to InitializeDatabase.

Reply:
Done.